### PR TITLE
feat: add 'deprecated' field for depreacted section generation in image docs

### PR DESCRIPTION
**Pull request by linostar**
- [x] I have signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] I am following the [CONTRIBUTING](https://github.com/canonical/oci-factory/blob/main/CONTRIBUTING.md) guidelines

*Ping the @canonical/rocks team.*

---

## Description
The documentation generator for rocks is missing some parts necessary to generate the `Deprecated channels & tags` section.

This PR provides the necessary values so that the generator can identify the deperacted tags and list them accordingly, using the expired end-of-life value of each track to determine whether it is depreacted or not.

### Related issues
OCI-Factory builds won't work if one image's track has an expired EOL value. Another PR to fix that issue will be out soon.

### CODE REVIEW

File: src/docs/generate_oci_doc_yaml.py
Comment (zhijie-yang) [inline src/docs/generate_oci_doc_yaml.py:1]: Some of your changes in `src/docs/generate_oci_doc_yaml.py` are not cherry-picked here.

**Review by zhijie-yang — CHANGES_REQUESTED** [review]

**Comment by linostar** [comment]
@clay-lake  @zhijie-yang This is ready for review.

The result can be seen on [Dockerhub's mock-rock page](https://hub.docker.com/r/rocksdev/mock-rock), where track `1.0-22.04` appears in the deprecated section because its EOL is `2024-05-01`.

**Comment by zhijie-yang** [comment]
image
image
Screenshots for the documentation on DockerHub.

**Review by zhijie-yang — APPROVED** [review]
Looks good to me. Thanks! @clay-lake wdyt?

**Review by clay-lake — APPROVED** [review]
Minor nit, but LGTM otherwise! Thanks.

File: src/image/release.py [hunk 1]
```diff
@@ -94,6 +94,16 @@
...
         tag_mapping_from_trigger[tag] = value
 
+# update EOL dates from upload dictionary
+for upload in image_trigger["upload"] or []:
+    for track, upload_release_dict in (upload["release"] or {}).items():
+        if track not in all_releases:
+            print(f"Track {track} will be created for the 1st time")
+            all_releases[track] = {}
+
+        if upload_release_dict and upload_release_dict.get("end-of-life"):
```
[the remark this answers is not in the document]
Comment (linostar) [reply src/image/release.py]: Applied your suggestion.

**Pull request merged**